src/module_3/model_selection_converted.md

```diff
@@ -0,0 +1,524 @@
+evaluate_classifier(best_estimator, X_val, y_val)
+```
+
+To gain a deeper understanding of the model's behavior, we examined the coefficients associated with each of the features:
+
+* It is observed that "ordered_before," "abandoned_before," and "global_popularity" exhibit relatively high positive coefficients, indicating their strong relationship with the predictive variable and their significance in predicting the response variable. 
+
+* The "price" and "days_since_purchase_product" features show negative coefficients, suggesting that as the number of days since the purchase or the price of the product increases, there is a decrease in the likelihood of purchasing the product.
+
+* Many variables such as "product type," "vendor," and "product type encoding" do not exhibit coefficients, possibly due to their categorical nature. This implies that the model may not be effectively leveraging these variables in its predictions.
+
+¿Possible solutions?
+
+- Feature engineering to explore alternative transformations to categorical variables
+- Incorporating domain knowledge to refine relevant features to enhance the model's predicitve performance
+
+
+```python
+coefficients = best_estimator.named_steps['reg'].coef_[0]
+feature_names = best_estimator.named_steps['preprocessor'].get_feature_names_out()
+
+coefficients_df = pd.DataFrame({'Feature': feature_names, 'Coefficient': coefficients})
+
+coefficients_df = coefficients_df.sort_values(by='Coefficient', ascending=False)
+
+print(coefficients_df)
+
+```
+
+It's noteworthy that we achieved an area under the ROC curve (AUC) over 80. However, it's important to interpret this metric cautiously due to the nature of our dataset. We have a considerable number of negative cases (non-purchases), which can heavily influence the ROC curve's false positive rate (FPR). In simpler terms, the AUC might appear high due to the large number of true negatives, but this doesn't necessarily reflect the model's effectiveness in correctly identifying positive instances (purchases). 
+
+
+```python
+plot_ROC_curve([best_estimator], X_val, y_val, ['Regresión logística Ridge'])
+```
+
+The area under the precision-recall (PR) curve for the Ridge Logistic Regression model is reported as 0.16. This indicates the model's performance in terms of precision and recall, which are crucial for our sales prediction task. However, the relatively low value of 0.16 suggests that the model may struggle to effectively balance precision and recall, resulting in suboptimal performance.
+
+
+```python
+plot_PR_curve([best_estimator], X_val, y_val, ['Regresión logística Ridge'])
+```
+
+<h2 style="text-align: left;">Threshold finding</h2>
+
+The precision-recall curve for different classification thresholds indicates that achieving a balance between precision and recall would require a threshold of approximately 0.1. At this threshold, the curve suggests that we could attain a recall of approximately 0.2 and a precision of 0.2.
```

Comment:
A balance doesn't not need to be the optimal for a given problem. It all depends on the business case.